blackjax/smc/partial_posteriors_path.py

```diff
@@ -0,0 +1,127 @@
+from typing import Callable, NamedTuple, Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from blackjax import SamplingAlgorithm, smc
+from blackjax.smc.base import update_and_take_last
+from blackjax.smc.from_mcmc import build_kernel as smc_from_mcmc
+from blackjax.types import Array, ArrayLikeTree, ArrayTree, PRNGKey
+
+
+class PartialPosteriorsSMCState(NamedTuple):
+    """Current state for the tempered SMC algorithm.
+
+    particles: PyTree
+        The particles' positions.
+    weights:
+        Weights of the particles, so that they represent a probability distribution
+    selector:
+        {Datapoints used to calculate the posterior the particles represent
```

Comment:
```suggestion
        Datapoints used to calculate the posterior the particles represent, a 1D boolean array to indicante which pariticles to include in the computation of the observed likelihood.
```

Is my understanding correct here? Also instead of `selector` I suggest renaming it to `particle_mask`

Reply:
hei! is not. If your likelihood is f(theta|d_1,...d_n) where d_i is a datapoint. The selector is used to choose
which ds get into the likelihood calculation.

Reply:
Recall that particles are just posterior samples. The datapoints here are from some dataset that defines the likelihood. Of course, if you are working with a potential-like function that has no points, this algorithm makes no sense.

Reply:
ah I see, would naming it `data_mask` be appropriate here?